olp-cpp-sdk-core/src/network2/curl/NetworkProtocolCurl.cpp

```diff
@@ -0,0 +1,1230 @@
+      }
+    }
+  }
+
+  // cURL teardown
+  curl_multi_cleanup(curl_);
+  curl_ = nullptr;
+
+#ifdef NETWORK_HAS_OPENSSL
+  // OpenSSL teardown
+  CRYPTO_set_id_callback(nullptr);
+  CRYPTO_set_locking_callback(nullptr);
+  gSslMutexes = nullptr;
+  ssl_mutexes_.reset();
+#endif
+
+#if (defined NETWORK_HAS_PIPE) || (defined NETWORK_HAS_PIPE2)
+  close(pipe_[0]);
+  close(pipe_[1]);
+#endif
+
+  // Handle completed messages
+  if (!completedMessages.empty()) {
+    for (auto& pair : completedMessages) {
+      pair.second(network2::NetworkResponse()
+        .WithRequestId(pair.first)
+        .WithStatus(static_cast<int>(ErrorCode::OFFLINE_ERROR))
+        .WithError("Offline"));
+    }
+  }
+  // Reset the memory tracker state.
+
+}
+
+bool NetworkProtocolCurl::IsStarted() const {
+  return state_ == WorkerState::Started;
+}
+
+bool NetworkProtocolCurl::Initialized() const { return IsStarted(); }
+
+bool NetworkProtocolCurl::Ready() {
+  if (!IsStarted()) {
+    return false;
+  }
+  std::lock_guard<std::mutex> lock(event_mutex_);
+  return std::any_of(
+      std::begin(handles_), std::end(handles_),
+      [](const RequestHandle& handle) { return !handle.in_use; });
+}
+
+size_t NetworkProtocolCurl::AmountPending() {
+  std::lock_guard<std::mutex> lock(event_mutex_);
+  return static_cast<size_t>(
+      std::count_if(std::begin(handles_), std::end(handles_),
+                    [](const RequestHandle& handle) { return handle.in_use; }));
+}
+
+
+NetworkStatus NetworkProtocolCurl::Send(NetworkRequest request,
+                                        std::shared_ptr<std::ostream> payload,
+                                        Network::Callback callback,
+                                        Network::HeaderCallback header_callback,
+                                        Network::DataCallback data_callback){
+
+  RequestId request_id {static_cast<RequestId>(RequestIdConstants::RequestIdMin)};
+  {
+    std::lock_guard<std::mutex> lock(event_mutex_);
+
+    request_id = next_request_id_;
+    if (next_request_id_ == static_cast<RequestId>(RequestIdConstants::RequestIdMax)) {
+      next_request_id_ = static_cast<RequestId>(RequestIdConstants::RequestIdMin);
+    } else {
+      next_request_id_++;
+    }
+  }
+
+  auto error_status = Send2(request, request_id, payload, std::move(header_callback), 
+                            std::move(data_callback), std::move(callback));
+
+  if (error_status == ErrorCode::NO_ERROR) {
+    return NetworkStatus(request_id);
+  }
+
+  return NetworkStatus(error_status);
+}
+
+
+ErrorCode NetworkProtocolCurl::Send2(
+    const NetworkRequest& request, int id,
+    const std::shared_ptr<std::ostream>& payload,
+    Network::HeaderCallback headerCallback, Network::DataCallback dataCallback,
+    Network::Callback callback) {
+  EDGE_SDK_LOG_TRACE(kLogTag, "send with id = " << id);
+
+  if (!IsStarted()) {
+    return ErrorCode::IO_ERROR;
+  }
+
+    const auto& config = request.GetSettings();
```

Comment:
```suggestion
  const auto& config = request.GetSettings();
```
Run clang-format on this file.